cmd/skopeo/copy.go

```diff
@@ -260,6 +261,15 @@ func (opts *copyOptions) run(args []string, stdout io.Writer) (retErr error) {
 		}
 	}
 
+	if destRef.Transport().Name() != "dir" {
+		if opts.destImage.dirForceCompression {
+			logrus.Warn("--dest-compress can only be used if the destination transport is 'dir'")
+		}
+		if opts.destImage.dirForceDecompression {
+			logrus.Warn("--dest-decompress can only be used if the destination transport is 'dir'")
+		}
+	}
```

Comment:
This is not quite the best place; `copy.go` otherwise doesn’t know or care about these two options at all.

I think this should happen somewhere in `imageDestOptions`, probably as a new method, something vaguely like `warnAboutIneffectiveOptions(destTransport types.ImageTransport)`, and probably with the `flagPrefix` recorded by `imageDestFlags` into `imageDestOptions.imageDestFlagPrefix`.

That would eventually allow similar warnings to be shared between `copy` and `sync` (or the other subcommands for reading images).